src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
```

Comment:
The use of `Objects` utility class is against the checklist guidelines. Consider implementing equality checks without using `Objects.equals`.